### ISSUE TITLE
Double prime

### ISSUE THREAD

**Issue by drlippman** [asciimath/asciimathml#88]
Can anyone think of a reason why AsciiMath shouldn't support a double-prime symbol, `f''` to translate to the u2033 character, like MathJax does for `$f''$` in TeX mode?

**Comment by saivan** [comment]
What would happen if we were writing some text, and somebody used a "<- ? How should that be handled?

**Comment by drlippman** [comment]
I don't think I'm understanding your question/concern. Can you clarify, please?

**Comment by drlippman** [comment]
I should have been more clear.  I was proposing that two single quotes be the double-prime symbol, not the double-quotes symbol used for surrounding text.  

Or are you saying that you currently use two single-quote symbols to mimic a double-quote symbol inside text, and are worried about the rendering of that changing?

**Comment by saivan** [comment]
yes sorry, I was saying that I currently use two single quotes inside text. But I reason that the rendering shouldn't change, but was wondering if we could escape certain symbols, then using two prime symbols would be unnecessary because we could just escape a double prime symbol.

**Comment by drlippman** [comment]
Gotcha.  It might be possible to support escaped quotes inside text - I'll take a look.  In the meantime, you might consider using the notation `text(hello "friend")` which works just fine.

One small concern with the double-prime:  FireFox's native rendering of it looks pretty bad:
![image](https://user-images.githubusercontent.com/190505/38784746-8b5a2374-40cb-11e8-8f0c-3de1da7e6150.png), 
though arguably not much worse than how it looks now: 
![image](https://user-images.githubusercontent.com/190505/38784752-a1397442-40cb-11e8-8190-dea928e11ee5.png).  

I think this is because AsciiMath is generating incorrect MathML for the primes.  It should be generating an `<msup>` according to the [specs](https://www.w3.org/Math/draft-spec/chapter7.html#chars.pseudo-scripts)

**Comment by saivan** [comment]
Ahh yes, that isn't nice at all :P 
I think it wouldn't be such a bad idea to add a symbol for as many primes as possible to be translated to an `<msup>`, so if some crazy soul wanted to do a fifth derivative that way (even though they shouldn't), you'd just translate it into the equivalent of `f^'''''`

**Comment by drlippman** [comment]
Yes, you're right, the existing script would require `f^('')(x)`.

I'm come up what I think is a better solution in the [issue88](https://github.com/asciimath/asciimathml/tree/issue88) branch.  If someone else could give it a spin and see if they can break it, that'd be much appreciated.

**Comment by drlippman** [comment]
One bummer (though this exists in the existing script) is that `f'(x)/2` gets displayed like `f' x/2` rather than like `{f'(x)}/2`.  That's because the "function hack" I put in place doesn't extend if modifiers are used, so `sin^2(x)/3` also displays as `sin^2 x/3` rather than `{sin^2(x)}/3` as might be desired.

**Comment by saivan** [comment]
It seems to be doing what I'd expect at the moment:

![image](https://user-images.githubusercontent.com/8543370/38845076-86788b74-4239-11e8-9265-3e68cf111e5e.png)

You're right though, if a user types `f'(x)/ 2`, it gives us this currentlly:

![image](https://user-images.githubusercontent.com/8543370/38845135-b418ddc2-4239-11e8-90d6-a107587c3f68.png)

But if you are making this last expression equivalent to `f^(')(x)/2`, it should work fine

![image](https://user-images.githubusercontent.com/8543370/38845303-42d91d10-423a-11e8-9d5d-d497a0c81482.png)

Even if you modify `f'''''(x)/2` to-> `f^(''''')(x)/2`, it still seems to work well

![image](https://user-images.githubusercontent.com/8543370/38845355-76da6b78-423a-11e8-9385-2c9d19ff6615.png)

So I'd say, if theres a prime immediately preceding a function, just replace it for `^(')` and you should be golden right?

**Comment by drlippman** [comment]
Whoops, I had forgotten to commit the actual code changes.  Please take another look.

And yes, you're right of course that `sin^2(x)/3` was displaying properly - I need to double-check things next time :)  I was able to find a way to extend the function hack to include primes.

**Comment by evictor** [comment]
I just use the Unicode double prime: [″](https://www.compart.com/en/unicode/U+2033).
